configure.ac

```diff
@@ -117,4 +114,32 @@ AS_IF([test "x$with_docs_only" = xyes],
     ]) # Docs only
 
 
+AC_ARG_ENABLE([debug],
+              AS_HELP_STRING([--enable-debug], [Enable extra debugging code (default is NO).]),
+              [], [enable_debug=no])
+
+if test $enable_debug = yes; then
+	AC_DEFINE([ENABLE_DEBUG], [1], [Enable debugging code])
+	CFLAGS="$CFLAGS -O0 -g3"
+	AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_TRACE_POLL], [Highest log level])
+else
+	AC_DEFINE([UCS_MAX_LOG_LEVEL], [UCS_LOG_LEVEL_INFO], [Highest log level])
+	CFLAGS="$CFLAGS -O3 -DNDEBUG"
+fi
+
+AM_CONDITIONAL([HAVE_UCX], [false])
+m4_include([m4/ucx.m4])
+CHECK_UCX
```

Comment:
Food for thought: UCG indirectly depends on UCX as well... what if it's on a different UCX? not urgent, but I need to cover that sooner or later

Reply:
Do you mean another UCX version ? Lets discuss it in WG, i think this is tricky.